### PR TITLE
Make python API the same as the other ones

### PR DESCRIPTION
**Pull request by leina05**
Updates to the FFI API. Split FFI functions among appropriate classes (`Polar`, `Query`, `QueryEvent`, `Error`). Upshot is that now the FFI `lib` functions only get called from `ffi.py`

### CODE REVIEW

File: languages/python/polar/ffi.py [hunk 1]
```diff
@@ -3,91 +3,119 @@
...
-        return json.loads(ffi.string(string).decode())
-    finally:
-        if not is_null(string):
-            lib.string_free(string)
+class QueryEvent:
+    def __init__(self, ptr):
+        self.ptr = ptr
 
+    def get(self):
+        return ffi.string(self.ptr).decode()
```
Comment (saolsen) [inline languages/python/polar/ffi.py:90]: do we ever actually call del on the QueryEvent to make sure it gets freed?

Comment (leina05) [reply languages/python/polar/ffi.py:90]: Oh good call. We don't but I thought the GC will do it?

Comment (saolsen) [reply languages/python/polar/ffi.py:90]: It wont always call __del__ when it goes out of scope like you’d want it to. Python does some weird object caching stuff. I think this came up before and we found some stackoverflow stuff about how __del__ shouldn’t be used for memory management like this. If you want an automatic thing you have to use a with statement but I think in this case this is fine we should just call del in the main event loop after handling an event.

Comment (leina05) [reply languages/python/polar/ffi.py:90]: Added

File: languages/python/polar/query.py [hunk 2]
```diff
@@ -36,9 +36,10 @@ def run(self):
         """Run the event loop and yield results."""
         assert self.ffi_query, "no query to run"
         while True:
-            event_str = self.ffi_query.next_event().get()
-            event = json.loads(event_str)
+            ffi_event = self.ffi_query.next_event()
+            event = json.loads(ffi_event.get())
             if event == "Done":
```
Comment (saolsen) [inline languages/python/polar/query.py:41]: I think this should happen on line 41 after the event is parsed. if it’s here it’ll only be called for done events

File: languages/python/polar/query.py [hunk 3]
```diff
@@ -40,15 +30,16 @@ def __init__(self, ffi_query, *, host=None):
...
     def run(self):
         """Run the event loop and yield results."""
         assert self.ffi_query, "no query to run"
         while True:
-            event_s = lib.polar_next_query_event(self.ffi_query)
-            event = ffi_deserialize(event_s)
+            ffi_event = self.ffi_query.next_event()
+            event = json.loads(ffi_event.get())
             if event == "Done":
+                del ffi_event
```
Comment (saolsen) [inline languages/python/polar/query.py:42]:
```suggestion
            ffi_event = self.ffi_query.next_event()
            event = json.loads(ffi_event.get())
            del ffi_event
            if event == "Done":
```

Comment (leina05) [reply languages/python/polar/query.py:42]: oh yeah you're totally right, I was thinking we still need it while the event loop is running but we don't

**Pull request merged**